spaceteam_proxy_handler.hpp

```diff
@@ -0,0 +1,78 @@
+#ifndef SPACETEAM_PROXY_HANDLER_HPP
+#define SPACETEAM_PROXY_HANDLER_HPP
+
+#include "request_handler.hpp"
+
+namespace http {
+namespace server {
+class SpaceteamProxyHandler : public RequestHandler {
+public:
+	SpaceteamProxyHandler();
+  virtual ~SpaceteamProxyHandler();
+	Status Init(const std::string& uri_prefix, const NginxConfig& config);
+	Status HandleRequest(const Request& request, Response* response);
+
+private:
+  // handler configuration
+	std::string uri;
+	std::string host;
+	std::string port;
+
+  // store the parsed response
+  std::string http_version;
+  int response_code;
+  using Headers = std::vector<std::pair<std::string, std::string>>;
+  Headers headers;
+  std::string body;
+
+  // the following are used for parsing the response
+
+  enum result_type { good, bad, indeterminate };
+
+  Status ParseResponse(const std::string& raw_response);
+  result_type consume(char input);
+

```

Comment:
Again, if you got this function from an outside boost library, can you please cite it at the top?